python/dgl/sparse/sparse_matrix.py

```diff
@@ -479,6 +479,71 @@ def is_diag(self):
         """Returns whether the sparse matrix is a diagonal matrix."""
         return self.c_sparse_matrix.is_diag()
 
+    def select(self, dim: int, ids: Union[tensor.Tensor, slice]):
+        """Returns a sub-matrix selected by a list of ids or by range.
+
+        Parameters
+        ----------
+        dim : int
+            The dim to select from matrix, should be 0 or 1.
+        ids : Union[tensor.Tensor, slice]
+            Tensor indicates the row ids to select from matrix.
+            NOTE: The ids can include duplicate values.
+            Slice indicates sub-matrix selected by range [start, end).
+
+        The function has not supported autograd.
+
+        Returns
+        -------
+        SparseMatrix
+            The sub-matrix only contains rows or columns selected.
+
+        Examples
+        --------
+
+        Case1: Select several rows by index.
```

Comment:
```suggestion
        Case 1: Select rows by IDs.
```

Reply:
Move this line after L508 `A = dglsp.spmatrix(indices, val)`, the sp matrix initiation code is shared across all cases.